bin/generators/plugins/install.js

```diff
@@ -0,0 +1,209 @@
+const eg = require('../../eg');
+const PluginInstaller = require('../../../lib/plugin-installer');
+
+module.exports = class extends eg.Generator {
+  constructor (args, opts) {
+    super(args, opts);
+
+    this.installer = PluginInstaller.create();
+
+    this.pluginOptions = null;
+
+    this.enablePlugin = false;
+    this.addPoliciesToWhitelist = false;
+
+    this.configureCommand({
+      command: 'install <package> [options]',
+      description: 'Install a plugin',
+      builder: yargs =>
+        yargs
+          .usage(`Usage: $0 ${process.argv[2]} install <package> [options]`)
+          .example(`$0 ${process.argv[2]} install express-gateway-plugin-url-rewrite`)
+          .option('n', {
+            alias: 'enable',
+            describe: 'Enable plugin in system config',
+            demandOption: false,
+            type: 'boolean'
+          })
+          .option('g', {
+            alias: 'update-gateway-config',
+            describe: 'Update policy whitelist in gateway config',
+            demandOption: false,
+            type: 'boolean'
+          })
+          .option('o', {
+            alias: 'option',
+            describe: 'User property in the form [-p \'foo=bar\']',
+            demandOption: false,
+            type: 'string'
+          })
+    });
+  }
+
+  initializing () {
+    this.enablePlugin = this.argv.n || null;
+    this.addPoliciesToWhitelist = this.argv.g || null;
+
+    let optionValues = [];
+
+    const argv = this.argv;
+    if (argv.o) {
+      optionValues = Array.isArray(argv.o) ? argv.o : [argv.o];
+    }
+
+    let hasInvalidProperty = false;
+
+    if (optionValues.length) {
+      this.pluginOptions = {};
+
+      const self = this;
```

Comment:
Is this still required? Arrow function should be automatically bound to the correct `this`, therefore you should be able to use that directly

Reply:
Nope.  Just an old habit.